src/libCli/ConnectionManager.hpp

```diff
@@ -14,137 +14,339 @@
 
 #pragma once
 
+#include <fstream>
+
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/security/credentials.h>
+
 #include <third_party/gRPC_utils/proto_reflection_descriptor_database.h>
 
 namespace cli
 {
     /// List of gRpc connection infomation
     typedef struct ConnList
     {
-       std::shared_ptr<grpc::Channel> channel = nullptr;
-       std::shared_ptr<grpc::ProtoReflectionDescriptorDatabase> descDb = nullptr;
-       std::shared_ptr<grpc::protobuf::DescriptorPool> descPool = nullptr;
+        std::shared_ptr<grpc::Channel> channel = nullptr;
+        std::shared_ptr<grpc::ProtoReflectionDescriptorDatabase> descDb = nullptr;
+        std::shared_ptr<grpc::protobuf::DescriptorPool> descPool = nullptr;
 
     } ConnList;
 
     /// Class to manage and resuse connection information, singleton pattern.
     class ConnectionManager
     {
-        public:
-            ConnectionManager(const ConnectionManager & ) = delete;
-            ConnectionManager& operator=(const ConnectionManager & ) = delete;
-        private:
-            ConnectionManager(){}
-            ~ConnectionManager(){}
-
-        public:
-            /// Only use a single connection instance
-            static ConnectionManager & getInstance()
+    public:
+        ConnectionManager(const ConnectionManager &) = delete;
+        ConnectionManager &operator=(const ConnectionManager &) = delete;
+
+    private:
+        ConnectionManager() {}
+        ~ConnectionManager() {}
+        int connectionStatus;
```

Comment:
better use `enum class`, not `int` for states. This allows better readability and more safety (prevents implicit conversions).